### PR TITLE
add CheckSemVer, CheckRootfsPath, CheckPlatform unit test

### PR DESCRIPTION
**Pull request by liangchenye**
Signed-off-by: liangchenye <liangchenye@huawei.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,5 +14,6 @@ script:
   - go vet -x ./...
   - $HOME/gopath/bin/golint ./...
   - $HOME/gopath/bin/git-validation -run DCO,short-subject -v -range ${TRAVIS_COMMIT_RANGE}
+  - go test -v ./...
   - make
```
Comment (wking) [inline .travis.yml]: Probably better to drop the explicit `vet` and `golint` calls and have:

```
script:
    - $HOME/gopath/bin/git-validation -run DCO,short-subject -v -range ${TRAVIS_COMMIT_RANGE}
    - make all test
```

File: Makefile [hunk 2]
```diff
@@ -37,7 +37,7 @@ $(RUNTIME_TOOLS_LINK):
 
 .PHONY: test .gofmt .govet .golint
```
Comment (wking) [inline Makefile]: You need to add `.gotest` here as well.

Comment (liangchenye) [inline Makefile]: Yes, I miss this.

File: validate/validate_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	}
+}
+
+func TestCheckSemVer(t *testing.T) {
+	cases := []struct {
+		val      string
+		expected bool
+	}{
+		{rspec.Version, true},
+		//FIXME: validate currently only handles rpsec.Version
```
Comment (liangchenye) [inline validate/validate_test.go]: We don't have the 'official supported version', so I prefer to keep this  and update this once the problem of supporting past version is fixed.

File: validate/validate_test.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	tmpBundle, err := ioutil.TempDir("", "oci-check-rootfspath")
+	if err != nil {
+		t.Fatalf("Failed to create a TempDir in 'CheckRootfsPath'")
+	}
+	defer os.RemoveAll(tmpBundle)
+
+	rootfsDir := "rootfs"
+	rootfsNonDir := "rootfsfile"
+	rootfsNonExists := "rootfsnil"
+	os.MkdirAll(filepath.Join(tmpBundle, rootfsDir), 0750)
```
Comment (liangchenye) [inline validate/validate_test.go]: I just used to set 0750 to all the created dir. I don't mind to change it to 0700. Update comes soon.

File: .travis.yml (hunk 1 above)
Comment (liangchenye) [inline .travis.yml]: how about  test first then try to compile?
- make test
- make all

Comment (wking) [inline .travis.yml]: That's fine with me.

**Comment by Mashimiao** [comment]
LGTM

**Comment by liangchenye** [comment]
ping @mrunalp @hqhq

File: validate/validate_test.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	if err != nil {
+		t.Fatalf("Failed to create a TempDir in 'CheckRootfsPath'")
+	}
+	defer os.RemoveAll(tmpBundle)
+
+	rootfsDir := "rootfs"
+	rootfsNonDir := "rootfsfile"
+	rootfsNonExists := "rootfsnil"
+	os.MkdirAll(filepath.Join(tmpBundle, rootfsDir), 0700)
+	os.Create(filepath.Join(tmpBundle, rootfsNonDir))
```
Comment (mrunalp) [inline validate/validate_test.go]: Missing error check.

Comment (liangchenye) [inline validate/validate_test.go]: fixed

File: validate/validate_test.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	tmpBundle, err := ioutil.TempDir("", "oci-check-rootfspath")
+	if err != nil {
+		t.Fatalf("Failed to create a TempDir in 'CheckRootfsPath'")
+	}
+	defer os.RemoveAll(tmpBundle)
+
+	rootfsDir := "rootfs"
+	rootfsNonDir := "rootfsfile"
+	rootfsNonExists := "rootfsnil"
+	os.MkdirAll(filepath.Join(tmpBundle, rootfsDir), 0700)
```
Comment (liangchenye) [inline validate/validate_test.go]: fixed

**Comment by liangchenye** [comment]
rebased, @mrunalp PTAL

**Comment by Mashimiao** [comment]
LGTM

File: Makefile (hunk 2 above)
Comment (wking) [inline Makefile]: Still missing a `.PHONY` entry for `.gotest`?

**Comment by Mashimiao** [comment]
need rebase

**Comment by liangchenye** [comment]
rebased, PTAL  @Mashimiao  @hqhq @mrunalp

**Comment by Mashimiao** [comment]
failed with CI

**Comment by liangchenye** [comment]
updated @Mashimiao 
Since we have validation/validation_test.go which is not UT, I added a UTDIRS to Makefile and now we will only do unit test under UTDIRS.

**Comment by Mashimiao** [comment]
LGTM

File: validate/validate_test.go [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+		{"invalid", false},
+	}
+
+	for _, c := range cases {
+		v := NewValidator(&rspec.Spec{Version: c.val}, "", false)
+		checkErrors(t, "CheckSemVer "+c.val, v.CheckSemVer(), c.expected)
+	}
+}
+
+func TestCheckPlatform(t *testing.T) {
```
Comment (hqhq) [inline validate/validate_test.go]: Platform is already removed from runtime-spec, so we don't need to add this is we're aiming at 1.0.

**Comment by liangchenye** [comment]
'Platform' UT is removed @hqhq 
I planed to remove the platform validation code this afternoon, but found it needs more time.
So I opened an issue to keep track of this. 
https://github.com/opencontainers/runtime-tools/issues/407

**Comment by hqhq** [comment]
LGTM

**Comment by Mashimiao** [comment]
LGTM

**Pull request merged**